gnd/src/main/java/com/google/android/gnd/ui/home/HomeScreenFragment.java

```diff
@@ -121,6 +123,15 @@ public void onCreate(@Nullable Bundle savedInstanceState) {
     projectSelectorViewModel = getViewModel(ProjectSelectorViewModel.class);
   }
 
+  private void onFeatureAdded(Optional<Feature> feature) {
+    feature.ifPresentOrElse(
+        viewModel::onAddFeature,
+        () -> {
+          // TODO: Show an error message to the user.
+          Timber.e("Couldn't add feature.");
```

Comment:
A more specific error message would be helpful here, but broader question - why would we ever call this with Optional.empty() in the first place?

Reply:
I agree that this is non-conventional. Ideally, we should return success state with result or error state with error message. I'll fix that. Thanks!